src/systems/systems/src/hydraulic/mod.rs

```diff
@@ -1622,6 +1625,185 @@ impl SimulationElement for LeakMeasurementValve {
+}
+impl FluidPhysics {
+    const MEAN_G_TRAP_CAVITY_TIME_DURATION_SECONDS: f64 = 20.;
+    const STD_DEV_G_TRAP_CAVITY_TIME_DURATION_SECONDS: f64 = 4.;
+
+    fn new() -> Self {
+        Self {
+            reference_point_cg: Vector3::default(),
+            fluid_cg_position: Vector3::new(0., -0.2, 0.),
+            fluid_cg_speed: Vector3::default(),
+
+            virtual_mass: Mass::new::<kilogram>(100.),
+            spring: SpringPhysics::new(),
+            anisotropic_damping_constant: Vector3::new(25., 20., 25.),
+
+            g_trap_is_empty: DelayedTrueLogicGate::new(Duration::from_secs_f64(
+                random_from_normal_distribution(
+                    Self::MEAN_G_TRAP_CAVITY_TIME_DURATION_SECONDS,
+                    Self::STD_DEV_G_TRAP_CAVITY_TIME_DURATION_SECONDS,
+                ),
+            )),
+        }
+    }
+
+    fn update(&mut self, context: &UpdateContext) {
+        self.update_speed_position(context);
+
+        self.g_trap_is_empty
+            .update(context, self.fluid_cg_position[1] > 0.);
+
+        // println!(
+        //     "current_fluid_cg {:.2} {:.2} {:.2} GTRAP EMPTY {:?} GAUGE {:.2}  USABLE {:.2}",
+        //     self.fluid_cg_position[0],
+        //     self.fluid_cg_position[1],
+        //     self.fluid_cg_position[2],
+        //     self.g_trap_is_empty.output(),
+        //     self.gauge_modifier().get::<ratio>(),
+        //     self.usable_level_modifier().get::<ratio>()
+        // );
+    }
+
+    fn update_forces(&mut self, context: &UpdateContext) -> Vector3<f64> {
+        let gravity_force = context.acceleration_plane_reference_filtered_ms2_vector()
+            * self.virtual_mass.get::<kilogram>();
+
+        let spring_force =
+            self.spring
+                .update_force_nm(context, self.reference_point_cg, self.fluid_cg_position);
+
+        let viscosity_damping = -self
+            .fluid_cg_speed
+            .component_mul(&self.anisotropic_damping_constant);
+
+        gravity_force + spring_force + viscosity_damping
+    }
+
+    fn update_speed_position(&mut self, context: &UpdateContext) {
+        let acceleration = self.update_forces(context) / self.virtual_mass.get::<kilogram>();
+
+        self.fluid_cg_speed += acceleration * context.delta_as_secs_f64();
+
+        self.fluid_cg_position += self.fluid_cg_speed * context.delta_as_secs_f64();
+    }
+
+    // Returns a coefficient of the actual level that will be seen by the gauge.
+    // Example: 0.5 means gauge reads half the actual level of the reservoir
+    fn gauge_modifier(&self) -> Ratio {
+        const LATERAL_BREAKPOINTS: [f64; 6] = [-1., -0.2, 0., 0.2, 0.4, 1.];
+        const LATERAL_MAP: [f64; 6] = [0.2, 0.95, 1., 1.05, 1.2, 1.3];
+
+        const LONGITUDINAL_BREAKPOINTS: [f64; 6] = [-1., -0.1, 0., 0.1, 0.4, 1.];
+        const LONGITUDINAL_MAP: [f64; 6] = [0.2, 0.98, 1., 0.98, 0.2, 0.2];
+
+        const VERTICAL_BREAKPOINTS: [f64; 6] = [-1., -0.1, 0., 0.1, 0.2, 1.];
```

Comment:
**suggestion**: Consider moving the consts closer to the calculation, e.g.:

```rust
const VERTICAL_BREAKPOINTS: [f64; 6] = [-1., -0.1, 0., 0.1, 0.2, 1.];
const VERTICAL_MAP: [f64; 6] = [1., 1., 0.7, 0.9, 1.2, 1.5];

let vertical_ratio = interpolation(
    &VERTICAL_BREAKPOINTS,
    &VERTICAL_MAP,
    self.fluid_cg_position[1],
);
```

It's a small thing, but it slightly reduces "context switching" as your eyes glide over the different axes. 😀

Reply:
Good point :)